src/txn/txn_ckpt.c

```diff
@@ -604,11 +604,9 @@ __checkpoint_prepare(WT_SESSION_IMPL *session, bool *trackingp, const char *cfg[
          * because recovery doesn't set the recovery timestamp until its checkpoint is complete.
          */
         if (txn_global->has_stable_timestamp) {
-            txn->read_timestamp = txn_global->stable_timestamp;
```

Comment:
Just confirming my understanding, is this the main change? Since we leave the read timestamp unset, we see the latest of each key?

Reply:
Yeah, this is the main change. We'll always see the latest committed update without a read timestamp.